run_wml_tests

```diff
@@ -68,9 +73,14 @@ def run_with_rerun_for_sdl_video(args, timeout):
     while sdl_retries < 10:
         # For compatibility with Ubuntu 16.04 LTS, this has to run on Python3.5,
         # so the capture_output argument is not available.
-        res = subprocess.run(args, timeout=timeout, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        if res.stderr.find(b"Could not initialize SDL_video") == -1:
-            return res
+        filename = get_output_filename(args)
+        res = subprocess.run(args, timeout=timeout, stdout=open(filename, "w"), stderr=subprocess.STDOUT)
+        retry = False
+        with open(filename, "r") as output:
```

Comment:
Can we be sure that the file handled opened for output has been flushed (or closed) by this point?

Reply:
How could `run()` still be writing to a file after it's returned?

Reply:
The subprocess could fork another process that keeps the FD open. Probably not something to worry about here.

Reply:
Resolved, a false belief on my part.

As Python uses reference counting, I think `stdout=open(...)` could leave the file open until some unspecified time in the future. However, I incorrectly believed that the `with open(...) as` syntax worked around that.